#goal def
int32 target_id
---
#result def
---
#feedback def
string last_state
